src/config.cc

```diff
@@ -201,7 +201,12 @@ namespace detail {
 			const auto has_dry_run_nodes = parsed_and_validated_envs.get(env_dry_run_nodes);
 			if(has_dry_run_nodes) { m_dry_run_nodes = *has_dry_run_nodes; }
 
+#if CELERITY_DIVERGENCE_CHECK
+			// divergence checker needs recording
+			m_recording = true;
+#else
 			m_recording = parsed_and_validated_envs.get_or(env_recording, false);
+#endif
```

Comment:
Should we print a warning that recording is being force-enabled here? What about the user explicitly setting `CELERITY_RECORDING=0`? cc @PeterTh @fknorr

Reply:
Agreed, maybe we should just get rid of it in a small follow-up PR.